### PR TITLE
In BackOffice, fixed problems for RTL languages

### LINKED ISSUE #26017: [BO] Mismatch between default-theme and new-theme for RTL languages
Hello,

#### Describe the bug

There is two themes for back-office, default and new-theme:
public_html/admin/themes/default
public_html/admin/themes/new-theme

Some pages works with default-theme and some pages works with new-theme.
I checked RTL version of default-theme and new-theme source codes (theme.rtlfix) and i found some problems:
1. Mismatch: Some CSS rules not exported from default-theme to new-theme (Rules that fixes RTL direction and Persian Fonts definition)
2. Outdated: Persian fonts defined in default-theme are outdated and need to update (Used: V18, Current: [V30](https://github.com/rastikerdar/vazir-font))

#### Expected behavior

Themes should to be synchronized to looks same everywhere in back-office.

#### Steps to Reproduce

(1) Steps to reproduce the behavior (**Arabic/Persian**):

1. Go to BO > Products
2. Edit a product
3. See categories list

(2) Steps to reproduce the behavior (**Persian**):
_(Persian language needs to be selected as default language for BO)_

1. Go to BO > Employees
2. Edit Admin account
3. Select Persian for language
4. See the main menu font (new-theme)
5. Go to BO > Dashboard
6. See the main menu font (default-theme)

Different font faces for different pages:

#### Additional information

* PrestaShop version: 1.7.8.0
* PHP version: 7.4

### PR DESCRIPTION
**Pull request by mparvazi**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fixes arrow direction. Adds support for material icons mirroring. Upgrade Vazir font to latest version. Unifies  the display between legacy and new theme. Removed duplicated CSS (Due to PR #26560).
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #26017
| How to test?      | Please see the images below.
| Possible impacts? | RTL version of the BO

# CSS modifications

## Font (Arabic/Persian)
https://github.com/rastikerdar/vazir-font

Go to BO > Dashboard

RTL font can be used in the login page.

## Header search
Fix dropdown menu width
Go to BO > Dashboard

## Category list
Fix arrows direction
Go to BO > Catalog > Catgories

## Catalog filter
Fix dropdown menu overflow
Go to BO > Catalog

## Pagination
Fix arrows direction
Go to BO > Catalog

## Popover
Fix arrow direction

## TinyMCE
Fix dropdown menu and popup position
Go to BO > Catalog > Edit a product

## History item (Product v2)
Fix history edit button position
Go to BO > Catalog > Edit a product > Edit a combinition

<br/>
<br/>

# Material icons

## Order navigation
Go to BO > Orders > Select an order

## Product combinations
Go to BO > Catalog > Edit a product > Edit a combination

## Product combinations (Product v2)
Go to BO > Catalog > Edit a product > Edit a combination

## Order preview
Go to BO > Orders > Open order preview

## Sidebar menu collapse
Go to BO > Dashboard

## Import data
Go to BO > Advanced Parameters > Import

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26700)

### CODE REVIEW

**Comment by mparvazi** [comment]
@NeOMakinG 

It's time to get rid of most of the RTL issues!

**Comment by mparvazi** [comment]
@NeOMakinG I made all the changes and It's ready for review.

**Review by NeOMakinG — DISMISSED** [review]
I don't see anything blocking this

Every RTL language users will love you honestly, huge work 👍

**Comment by NeOMakinG** [comment]
@Julievrz I set Wording ✅ because there are no wording changes!

**Comment by mparvazi** [comment]
@NeOMakinG 

I always had problem with material icons mirroing in my designs.
Your comment really helped me:
https://github.com/PrestaShop/PrestaShop/pull/26147#issuecomment-957208845
That was a smart and easy solution for RTL.
Thank you.

**Comment by mparvazi** [comment]
@khouloudbelguith 

A lot of changes has been made.
We need your keen eye here!

**Comment by NeOMakinG** [comment]
> #26147 (comment)

You're welcome, win-win 🚀

**Comment by Julievrz** [comment]
> @Julievrz I set Wording ✅ because there are no wording changes!

Ok, thank you! @NeOMakinG

**Review by Progi1984 — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @mparvazi,

First of all, thanks for your PR and for the huge fix :tada: 

But I found some issues should be fixed:

**Issue1**

1. Go to BO > Orders > Add new Order
2. Choose a customer > choose a product
3. In the summary tab
4. Click "More actions"
5. See error 

image

**Without PR** => ok
image

**Issue2**
I have with & without PR
1. Go to BO > Catalog > Products > Edit a product with combinations
2. In the combinations Tab > Click on bulk actions > Click on the Availability date 
3. check the error 
image
4. same error if you edit a combination 
image
5. same issue if you edit a vector product 
image


**Issue3**
I have an issue with & without PR
1. Go to BO > Design > Positions page
2. click on the three points to unhook a module
3. See an error 
image

**Issue4**
1. In the BO > Dashboard page
2. Use a mobile
3. see an error 
image

**Without PR** 

image

Thanks!

**Comment by mparvazi** [comment]
@khouloudbelguith  Thanks for your feedback.

That's why you're great at QA!

I'll work on it.

**Comment by mparvazi** [comment]
@khouloudbelguith Fixed!

## Issue#1


## Issue#2


## Issue#3


## Issue#4
I think it is not related to this PR, and that's a problem with responsive device tool when interpreting the flex elements in RTL.
By the way it's a emulator that interoperates the page elements in a different way of rendering pages by browsers.
 
issue4



## Mobile testing
I tested with my phone at localhost.

### Safari


### Firefox




### Chrome

**Review by NeOMakinG — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by mparvazi** [comment]
https://user-images.githubusercontent.com/85633460/143931935-e8762da5-8cb2-4a64-b124-53f03142bc61.MP4

**Comment by NeOMakinG** [comment]
@mparvazi damn, there is a conflict, can you rebase? :)

**Comment by mparvazi** [comment]
@NeOMakinG I always have problem with rebase! I prefere to make a new PR. Is it ok?

**Comment by mparvazi** [comment]
@NeOMakinG I used "Resolve conflicts" option. Is it enough?

**Comment by NeOMakinG** [comment]
@mparvazi Damn no, you now have a merge commit inside the PR

Let me take a look and I'll send you every command I use

**Comment by NeOMakinG** [comment]
@mparvazi fixed it

I did: 

```
git pull upstream develop
git checkout mparvazi/rtl-fix --track
git rebase develop
// Resolve every conflicts then git add them
git add ../../../src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Attribute/index.html.twig
git rebase --continue
git push mparvazi rtl-fix -f
```

In you case `mparvazi` remote is `origin` and you should add the `upstream` remote which is the prestashop official repository

**Review by NeOMakinG — DISMISSED** [review]

**Comment by mparvazi** [comment]
@NeOMakinG I've got errors! 
What about creating a new PR?

**Comment by mparvazi** [comment]
@Progi1984 @matks 

Merging this PR has become a dream for me :)

**Comment by NeOMakinG** [comment]
> @Progi1984 @matks
> 
> Merging this PR has become a dream for me :)

Don't worry, your dream will come true asap 😆

**Comment by khouloudbelguith** [comment]
Hi @mparvazi,

I found some bugs.
Could you please confirm if those issues are for the scope of this PR:

**First issue**

1. Go to BO > Advanced Parameters > Enable Experimental feature
2. Go to BO > Catalog > Edit Product id_product = 1 (demo) V2
3. In the combinations tab > Filter by > Click on Size & Click on Color 
4. See error 

image
image

English lanaguge => ok
image


**Second issue**
1. Go to BO > International > Tax rules page & edit a tax rule
2. See error 
image

It should be 
image



**Third issue**
1. Go to BO > Design > Theme & Logo page > Advanced Customization Tab
2. Click on Upload child theme
3. See error 
image


English language
image

PS: about the dashboard using a **mobile**  => I use a browser Stack to test mobile
1. IOS&os_version=14.0&device_browser=safari
image

2. os=android&os_version=11.0&device=Samsung+Galaxy+S21+Ultra&device_browser=chrome
image


Thanks!

**Comment by mparvazi** [comment]
@khouloudbelguith Fixed!

## First issue
Screenshot 2021-12-01 at 17-51-27 المنتجات • prestashop

Screenshot 2021-12-01 at 17-51-33 المنتجات • prestashop


## Second issue
This page designed in the legacy theme. It must be fixed in another PR, because it isn't just related to RTL.
The dropdown menu always will be shown at the bottom and the caret icon (▾) direction doesn't change.

**RTL**
Screenshot 2021-12-01 at 18-00-13 القواعد الضريبية تحرير US-AL Rate (4%) • prestashop

**LTR**
Screenshot 2021-12-01 at 18-07-41 Tax Rules Edit US-AL Rate (4%) • prestashop

## Third issue
Screenshot 2021-12-01 at 18-19-05 Advanced Customization • prestashop

**Comment by mparvazi** [comment]
## Mobile test

Chrome dev tools
Screenshot from 2021-12-02 00-47-43

**Comment by mparvazi** [comment]
## Mobile test

The header overflow fixed, I hope! 

**Chrome DevTools (new-theme)**



**Chrome DevTools (default-theme)**



**Chrome iOS**

**Comment by mparvazi** [comment]
Please check the latest changes I made carefully, It's 2AM and I don't know how I found the problem.

**Comment by mparvazi** [comment]
@khouloudbelguith The product v2 page is WIP. I think it's better to fix RTL issues for this page after it's completed.

**Review by NeOMakinG — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by mparvazi** [comment]
Me when waiting for merge this PR :|

**Comment by NeOMakinG** [comment]
> Me when waiting for merge this PR :|
> 
> 

QA can be really busy sometimes... :/

**Comment by matks** [comment]
@mparvazi QA team does an outstanding job, they have a deep and thorough analysis of the software and use it on each PR to detect possible regressions. PrestaShop is a very big software, only skilled people can QA it*. But as the software is big, as the need for knowledge and accuracy is high (because they are the main people finding bugs in the code we bring), it takes a lot of time to validate each PR.

Between speed and quality, I choose quality :)

*Unfortunately it also means it's long for someone to be onboarded to become a QA analyst on this project

**Comment by marwachelly** [comment]
Hello,
Thanks for your fix but I detected the following bug caused after RTL activaion:

1. Go to catalog > Products> Edit product > Pricing > currency is not displayed
image
image

2. Go to orders then resize the window to 1024 x 886

image
image

3. Go to customer service > order messages > click on delete > unecessery bar is displayed
image
image

Could you please check,
Thanks,

**Comment by mparvazi** [comment]
Hello @marwachelly 
Thanks for your feedback!

**First issue:**
> Go to catalog > Products> Edit product > Pricing > currency is not displayed

For product page in RTL the currency symbol element inside HTML is empty `<span class="input-group-text"></span>`
It's not related to this PR (This is a Bug for RTL languages but cannot be solved with CSS)

**Second issue:**
> Go to orders then resize the window to 1024 x 886

It can be fixed!

**Third issue:**
> Go to customer service > order messages > click on delete > unecessery bar is displayed

It generally happens for dropdown menu at the last row of the grid table.
For example `Go to Customers > Addresses`
The placement of the dropdown menu calculate by jQuery as an inline style (e.g. `translate3d(-12px, -39px, 0px)`)
It's not related to this PR (This is a Bug for RTL languages but cannot be solved with CSS)

**Comment by mparvazi** [comment]
About first issue and third issue that has been found by @marwachelly but i didn't fix here:

In this PR I tried to focus on CSS problems for RTL.

Some of the problems in RTL originally is not related to CSS, however can be fixed by CSS tricks, but it may cause mismatch between LTR and RTL appearance.

I think these problems should be solved from where they occurred (core or scripts) so it's better to create new issues for these problems separately.

**Review by NeOMakinG — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
@marwachelly We need to stop asking more and more on this PR, @mparvazi is doing well and works on fixing things you encountered, but at first, this PR was created to fix `some` bugs that he found, we could have add every other bug to another issue and work on it inside another PR

Luckily @mparvazi didn't give up on this work, but I really feel like not every contributors would have been that enduring!

Let's add every other bugs not linked to bugs fixed in this PR to another issue and go further :)

**Comment by marwachelly** [comment]
> @marwachelly We need to stop asking more and more on this PR, @mparvazi is doing well and works on fixing things you encountered, but at first, this PR was created to fix `some` bugs that he found, we could have add every other bug to another issue and work on it inside another PR
> 
> Luckily @mparvazi didn't give up on this work, but I really feel like not every contributors would have been that enduring!
> 
> Let's add every other bugs not linked to bugs fixed in this PR to another issue and go further :)

Thanks @NeOMakinG and @mparvazi  for your work :) 
So PR will be validated and I will create new tickets for all issues detected.

**Comment by mparvazi** [comment]
@NeOMakinG  @marwachelly

The problem for dropdown overflow at grid table is related to `popper.js`.
When the dropdown is inside a box in a responsive design, the position of dropdown will be calculated incorrectly in RTL.



I thought it should be fixed by `popper.js` settings, but it can be fixed easily by CSS.

Sorry! the PR approved and check done, but I decided to solve the problem now, because I may forget later 🙃

**Comment by mparvazi** [comment]
@marwachelly 

You can test the last change that I've made in grid tables:
Catalog > Categories
Customer > Addresses
International > Languages
Design > Pages

**Comment by marwachelly** [comment]
> @marwachelly
> 
> You can test the last change that I've made in grid tables: Catalog > Categories Customer > Addresses International > Languages Design > Pages

In this case we should wait the review from dev side. So PR is waiting for dev again.

File: admin-dev/themes/new-theme/public/theme.rtlfix [hunk 1]
```diff
@@ -183,6 +183,6 @@ ul.category-tree .more > .radio:before,
 }
 
 /* Fix dropdown overflow inside grid table */
-.grid-table {
-  overflow-x: auto;
```
Comment (mparvazi) [inline admin-dev/themes/new-theme/public/theme.rtlfix]: Worked fine for Firefox but had problem in Chrome.

File: admin-dev/themes/new-theme/public/theme.rtlfix [hunk 2]
```diff
@@ -183,6 +183,6 @@ ul.category-tree .more > .radio:before,
...
+.table-responsive {
+  overflow-y: hidden;
```
Comment (mparvazi) [inline admin-dev/themes/new-theme/public/theme.rtlfix:187]: Works fine for Firefox and Chrome.
The `grid-table` is inside `table-responsive`
I don't know why hiding vertical overflow can fixes horizontal overflow problem!

Comment (NeOMakinG) [inline admin-dev/themes/new-theme/public/theme.rtlfix:187]: Can you check the dropdown of the last line inside a table? I think it can cause some bugs!

Comment (mparvazi) [reply admin-dev/themes/new-theme/public/theme.rtlfix:187]: https://watch.screencastify.com/v/dGcMMM4lxpPBBGfipHIF

Comment (mparvazi) [reply admin-dev/themes/new-theme/public/theme.rtlfix:187]: Mobile test:
https://user-images.githubusercontent.com/85633460/145776442-3218ad2b-adfe-4c5f-b4c3-d421e966e4c9.mp4

**Comment by mparvazi** [comment]
@NeOMakinG 

I always had problem to solve dropdown overflow for last row in the grid table.
I wanted to test `overflow-x: hidden;` but by mistake I wrote `overflow-y: hidden;` and it works fine!

Chrome (DevTools):

https://watch.screencastify.com/v/dGcMMM4lxpPBBGfipHIF

Chrome (iOS):

https://user-images.githubusercontent.com/85633460/145776573-4391b634-7f9c-44d1-aa65-d9d21684f799.mp4

**Comment by NeOMakinG** [comment]
> @NeOMakinG
> 
> I always had problem to solve dropdown overflow for last row in the grid table. I wanted to test `overflow-x: hidden;` but by mistake I wrote `overflow-y: hidden;` and it works fine!
> 
> Chrome (DevTools):
> 
> https://watch.screencastify.com/v/dGcMMM4lxpPBBGfipHIF
> 
> Chrome (iOS):
> 
>  RPReplay_Final1639381737.mp4

It works in case you've a few items, hope that we don't have any tables with 5+ items :p

**Comment by mparvazi** [comment]
@NeOMakinG 

> It works in case you've a few items, hope that we don't have any tables with 5+ items :p

OK, I'll test with more items.

**Comment by mparvazi** [comment]
@NeOMakinG 

BO > Design > Pages

Chrome (Desktop):

https://watch.screencastify.com/v/SGbaSTW4jbhL72tFLqoI


Chrome (Mobile):


https://user-images.githubusercontent.com/85633460/145786569-1a5581de-b2f8-49fe-bb1d-25789445faae.MP4

**Comment by mparvazi** [comment]
Btw I don't know why `overflow-y: hidden;` works fine here!

I thought the problem is related to the custom font, then I deleted the font-face but problem still remained.
So I think this strange behavior is related to responsive display in RTL, because when I remove `table-responsive` class, there is no problem. 

If you're not interested, we can remove it from this PR, and I'll try to find another solution by creating a new issue for this problem only.

**Comment by NeOMakinG** [comment]
> Btw I don't know why `overflow-y: hidden;` works fine here!
> 
> I thought the problem is related to the custom font, then I deleted the font-face but problem still remained. So I think this strange behavior is related to responsive display in RTL, because when I remove `table-responsive` class, there is no problem.
> 
> If you're not interested, we can remove it from this PR, and I'll try to find another solution by creating a new issue for this problem only.

It should be fine, just let it and see if we've some feedbacks later, it's working atm

**Comment by NeOMakinG** [comment]
> > Btw I don't know why `overflow-y: hidden;` works fine here!
> > I thought the problem is related to the custom font, then I deleted the font-face but problem still remained. So I think this strange behavior is related to responsive display in RTL, because when I remove `table-responsive` class, there is no problem.
> > If you're not interested, we can remove it from this PR, and I'll try to find another solution by creating a new issue for this problem only.
> 
> It should be fine, just let it and see if we've some feedbacks later, it's working atm

Btw I just seen that you tested with more items in the table, but not inside the dropdown, add 6-7 items in the latest dropdown to see how it react :p

**Comment by mparvazi** [comment]
@NeOMakinG 

> Btw I just seen that you tested with more items in the table, but not inside the dropdown, add 6-7 items in the latest dropdown to see how it react :p

OK, I got it!

**Comment by mparvazi** [comment]
@NeOMakinG 

https://watch.screencastify.com/v/KysDdfs5GKSTxMuzV0p0

https://user-images.githubusercontent.com/85633460/146017941-29fc9ecc-b8e8-440e-97a8-938ff3254c9b.MP4

**Review by NeOMakinG — DISMISSED** [review]

**Comment by NeOMakinG** [comment]
@mparvazi great, bootstrap is calculating right!

File: admin-dev/themes/default/template/controllers/login/header.tpl [hunk 3]
```diff
@@ -49,6 +49,6 @@
 		{/if}
     <script type="text/javascript" src="../js/admin/login.js?v={$smarty.const._PS_VERSION_|escape:'html':'UTF-8'}"></script>
 	</head>
-	<body class="ps_back-office bootstrap">
+	<body class="lang-{$iso_user}{if $lang_is_rtl} lang-rtl{/if} ps_back-office bootstrap">
```
Comment (eternoendless) [inline admin-dev/themes/default/template/controllers/login/header.tpl]: Is this part really necessary?
```suggestion
	<body class="{if $lang_is_rtl} lang-rtl{/if} ps_back-office bootstrap">
```

Comment (mparvazi) [reply admin-dev/themes/default/template/controllers/login/header.tpl]: > Is this part really necessary?

Not really, I just copied from new theme :)
`{if $lang_is_rtl}lang-rtl{/if}` is enough.

Comment (NeOMakinG) [reply admin-dev/themes/default/template/controllers/login/header.tpl]: Maybe we should remove it from new-theme too?

**Review by sowbiba — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by marwachelly** [comment]
Hello @mparvazi,

Thank you for you work. I tested the PR and I noticed the following case :

- the issue in BO > Design > Pages using mobile is fixed 

https://watch.screencastify.com/v/31XPvFfeNMV2tcheAtFS

- The issue In the BO > Dashboard page Using a mobile is also fixed


- The drop-down issue is not resolved when you click on delete in the last row

Catalog > Categories
image


Customer > Addresses
https://watch.screencastify.com/v/TucV8hDBl6IEyrZRdkhv

International > Languages
https://watch.screencastify.com/v/KbqIgyFkd4TOLEGEM2mi

Design > Pages
https://watch.screencastify.com/v/wTS9QjFyklypSswfy98H

Could you please check?
Thanks!

**Comment by mparvazi** [comment]
Hello @marwachelly 

Please clear the browser cache.

Use inspector to see if this style is added to `table-responsive`
```
.table-responsive {
    overflow-y: hidden;
}
```

146803329-9741e20f-d09d-48f7-861b-09b2883317bf


**OR** maybe it's related to `theme_rtl.css` file:
1. Delete the file PS/admin-dev/themes/new-theme/public/`theme_rtl.css`
2. Go to BO > Localization > Languages 
3. Disable the "RTL language" which you are using
3. Enable that "RTL language" again
4. Now `theme_rtl.css` is generated

**Comment by mparvazi** [comment]
The CSS rules for RTL in this PR added in theme.rtlfix

Content of this file will be added to theme_rtl.css just once when for first time a RTL language is will be add to shop.
So each time an update will be commited to theme.rtlfix, then we must create theme_rtl.css again.

theme_rtl.css = theme.css + theme.rtlfix

**Comment by marwachelly** [comment]
Hello,

The drop-down issue is still not resolved when you click on delete in the last row and following the steps bellow: 

1. clear the browser cache.
2. Delete the attached selected files from PS/admin-dev/themes/new-theme/public/
image

3. Go to BO > Localization > Languages
4. Disable the "RTL language" which you are using
5. Enable that "RTL language" again
6. Now deleted files are generated

https://watch.screencastify.com/v/M4CmMKS2spHFEKClIQXv

Could you please check?

**Comment by mparvazi** [comment]
I tried to test again with fresh install:

Screenshot from 2022-01-05 11-06-53

**Test page:**
BO > Catalog > Products
The dropdown menu at last row.

**Firefox:**

Screenshot from 2022-01-05 11-32-13


**Chrome:**

Screenshot from 2022-01-05 11-40-12

**Safari:**



It works fine for all table grids in Firefox and Chrome.

@NeOMakinG Did I miss something?

**Comment by NeOMakinG** [comment]
@mparvazi try to disable the cache inside the network tabs, and refresh the page with the console opened ?

Overwise It might be a cache issue from @marwachelly, I'll try to test it my self

**Comment by mparvazi** [comment]
@NeOMakinG 

Test with empty cache:

Screenshot from 2022-01-06 09-05-41

**Comment by NeOMakinG** [comment]
image
image

Looks also fine to me

**Comment by marwachelly** [comment]
Hello,

I observed the problem when I installed the shop with english language. But when installing the shop with an RTL language the issue is fixed. PR is QA approved.

Thank you ,

**Comment by mparvazi** [comment]
Ty @marwachelly

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @mparvazi

**Pull request merged**